File: go-algorithm/leetcode/problem413.go
package main

// 413. Arithmetic Slices
// https://leetcode.com/problems/arithmetic-slices/

func numberOfArithmeticSlices(nums []int) int {
	n := len(nums)
	if n < 3 {
		return 0
	}
	ans := 0
	for i := 0; i < n-2; i++ {
		d := nums[i+1] - nums[i]
		j := i + 2
		for j < n && nums[j]-nums[j-1] == d {
			j++
		}
		if j-i >= 3 {
			ans += j - i - 2
		}
	}
	return ans
}

type Person struct {
	Name string
}

type School struct {
	Extension *string
}

func change(person *Person) {
	person.Name = "BBB"
}

File: python-algo/leetcode/problem_413.py
"""413. Arithmetic Slices
https://leetcode.com/problems/arithmetic-slices/
"""
from typing import List


class Solution:
    def number_of_arithmetic_slices(self, nums: List[int]) -> int:
        i = 0
        ans = 0
        while i < len(nums) - 2:
            j = i + 1
            diff = nums[j] - nums[i]
            while j < len(nums) and nums[j] - nums[j - 1] == diff:
                j += 1
            n = j - i
            if n >= 3:
                ans += (n - 1) * (n - 2) // 2
                i = j - 1
            else:
                i += 1
        return ans
